README.md

```diff
@@ -1,486 +1,679 @@
+
+## `@logged`
+
+The `@logged` decorator logs a console message when a method starts and finishes. Many other popular decorators will also want to wrap a function, e.g., `@deprecated`, `@debounce`, `@memoize`, etc.
 
 Usage:
 
 ```mjs
-import { @logged } from "./logged.mjs";
+import { logged } from "./logged.mjs";
 
 class C {
   @logged
-  method(arg) {
+  m(arg) {
     this.#x = arg;
   }
 
   @logged
   set #x(value) { }
 }
 
-new C().method(1);
-// starting method with arguments 1
+new C().m(1);
+// starting m with arguments 1
 // starting set #x with arguments 1
 // ending set #x
-// ending method
+// ending m
 ```
 
-`@logged` can be implemented in JavaScript in terms of built-in decorators: The `@logged` decorator is defined in terms of the `@wrap` built-in decorator, which takes a function as an argument. The method is passed through this function to get the method which is finally present on the class. `@wrap` is similar to core-decorators' [`@decorate`](https://www.npmjs.com/package/core-decorators#decorate) decorator.
+`@logged` can be implemented in JavaScript as a decorator. Decorators are functions that are called with an argument containing what's being decorated. For example:
+- A decorated method is called with the method being decorated
+- A decorated getter is called with the getter function being decorated
+- A decorated setter is called with the setter function being decorated
+
+(Decorators are called with a second parameter giving more context, but we don't need those details for the `@logged` decorator.)
+
+The return value of a decorator is a new value that replaces the thing it's wrapping. For methods, getters and setters, the return value is another function to replace that method, getter or setter.
 
 ```mjs
 // logged.mjs
 
-export decorator @logged {
-  @wrap(f => {
-    const name = f.name;
-    function wrapped(...args) {
-      console.log(`starting ${name} with arguments ${args.join(", ")}`);
-      f.call(this, ...args);
-      console.log(`ending ${name}`);
-    }
-    wrapped.name = name;
-    return wrapped;
-  })
+export function logged(f) {
+  const name = f.name;
+  function wrapped(...args) {
+    console.log(`starting ${name} with arguments ${args.join(", ")}`);
+    const ret = f.call(this, ...args);
+    console.log(`ending ${name}`);
+    return ret;
+  }
+  wrapped.name = name;
+  return wrapped;
 }
 ```
 
-In the above example, the *composed decorator* `@logged` is defined to expand out into a call of the `@wrap` decorator with a particular fixed callback.
+This example roughly "desugars" to the following (i.e., could be transpiled as such):
 
-### `@defineElement`
+```js
+let x_setter;
+
+class C {
+  method(arg) {
+    this.#x = arg;
+  }
+
+  static #x_setter(value) { }
+  static #_ = (x_setter = C.#x_setter, void 0);
+  set #x(value) { return x_setter.call(this, value); }
+}
+
+C.prototype.method = logged(C.prototype.method, { kind: "method", name: "m", isStatic: false });
```

Comment:
Why is the method's name is "m", when it is called "method"?

Reply:
...because I didn't test any of this code! Thanks for pointing out the error; will fix.